We conducted a large-scale, statistically powered, meta-analysis of exercise adaptations in human skeletal muscles, integrating epigenetic, transcriptomic, transcription factors, and proteomic data across 12 independent cohorts comprising over 1000 participants and 2340 human muscle samples. Our study identified distinctive signatures associated with maximal oxygen consumption (VO2max), and identified five genes robustly intersecting multi-OMIC layers. Notably, transcription factors predominantly functioned as activators across these layers, regulating expression of target genes irrespective of whether DNA methylation levels were low or high, indicating a synergistic effect between TFs and the methylome. Analysis of distinct exercise modalities (aerobic and resistance exercise) revealed unique gene pathways, contrasting with patterns observed in inactivity (muscle disuse) studies. These findings offer a comprehensive understanding of exercise and modality-specific adaptations, shedding light on muscle health and the molecular mechanisms associated with cardiorespiratory fitness, aging, and disease prevention.
